### DISCUSSION TITLE
How to retrieve the states of the devices programatically?

### DISCUSSION THREAD

**Discussion by wsw70** [Koenkk/zigbee2mqtt#8183] [General]
I know how to retrieve a (zipped) JSON file with the status of all devices (sigbee2mqtt → Settings → Tools → Download states).

Is it possible to access these data via an API, or otherwise download the JSON file?

**Comment by Koenkk** [comment]
Only via mqtt, if you set `retain: true` the latest state can always be found on `zigbee2mqtt/your_device` (https://www.zigbee2mqtt.io/information/configuration.html#all-devices)

**Reply by wsw70** [reply]
Thank you. I was hoping for an API access because I saw that `https://zigbee2mqtt.examlple.com/#/settings/bridge` provides (among others) the list of devices, and then `https://zigbee2mqtt.example.com/#/device/<deviceId>/info` gives (again, among others) the last seen timestamp.

I wanted to avoid scrapping the data from the site (especially that the last seen time I provided in a relative format).

The main reason why I want to avoid MQTT is that this is for monitoring purposes (I use my own monitoring system for my home appliances and services)  and I do not want to have dependencies (not to mention that an event-based solution such as MQT is not ideal for monitoring purposes)
